src/main/java/com/google/cloud/genomics/utils/GenomicsUtils.java

```diff
@@ -56,8 +60,61 @@
     }
     return output;
   }
+
+  /**
+   * Gets the ReferenceSetId for a given readGroupSetId using the Genomics API.
+   *
+   * @param readGroupSetId The id of the readGroupSet to query.
+   * @param auth The OfflineAuth for the API request.
+   * @return The referenceSetId for the redGroupSet (which may be null).
+   * @throws IOException
+   * @throws GeneralSecurityException
+   */
+  public static String getReferenceSetId(String readGroupSetId, GenomicsFactory.OfflineAuth auth)
+      throws IOException, GeneralSecurityException {
+    Genomics genomics = auth.getGenomics(auth.getDefaultFactory());
+    ReadGroupSet readGroupSet = genomics.readgroupsets().get(readGroupSetId)
+        .setFields("referenceSetId").execute();
+    return readGroupSet.getReferenceSetId();
+  }
   
   /**
+   * Gets the CoverageBuckets for a given readGroupSetId using the Genomics API.
+   *
+   * @param readGroupSetId The id of the readGroupSet to query.
+   * @param auth The OfflineAuth for the API request.
+   * @return The list of reference bounds in the variantSet.
+   * @throws IOException
+   * @throws GeneralSecurityException
+   */
+  public static List<CoverageBucket> getCoverageBuckets(String readGroupSetId, GenomicsFactory.OfflineAuth auth)
+      throws IOException, GeneralSecurityException {
+    Genomics genomics = auth.getGenomics(auth.getDefaultFactory());
+    // Not using a Paginator here because requests of this form return one result per
```

Comment:
This don't agree that this is a safe assumption. If there is already a paginator I would strongly prefer it were used. Otherwise, you should at least throw an exception if nextPageToken is set.